### PR TITLE
Feature/801 refactor state

### PR DESCRIPTION
**Pull request by fzuray**

### CODE REVIEW

File: src/warzone/src/warzone/service/CommandService.java [hunk 1]
```diff
@@ -15,23 +15,24 @@
 	private static CommandService COMMAND_SERVICE;
 	
 	private GameContext d_gameContext;
+	private GameEngine d_gameEngine;	
 
 	/**
 	 * The constructor of the class.
 	 * @param p_gameContext initiating the game context of the current object
```
Comment (amyhcy) [inline src/warzone/src/warzone/service/CommandService.java]: param is p_gameEngine

File: src/warzone/src/warzone/service/CommandService.java [hunk 2]
```diff
@@ -15,23 +15,24 @@
...
 	 */
-	private CommandService(GameContext p_gameContext) {
-		d_gameContext = p_gameContext;
+	private CommandService(GameEngine p_gameEngine) {
+		d_gameEngine = p_gameEngine;
 	}	
 	
 	/**
 	 * This method will return command service object from game context.
 	 * @param p_gameContext the needed game context
```
Comment (amyhcy) [inline src/warzone/src/warzone/service/CommandService.java:30]: param is p_gameEngine

File: src/warzone/src/warzone/service/GameEngine.java [hunk 3]
```diff
@@ -36,25 +56,47 @@ public static GameEngine getGameEngine(GameContext p_gameContext) {
...
-		GameContext l_gameContext = GameContext.getGameContext();
-		RouterService l_routerService =  RouterService.getRouterService(l_gameContext);
-		CommandService l_commandService =  CommandService.getCommandService(l_gameContext);
-		
+	private Phase d_gamePhase ;
+	
+	public Phase getPhase() {
+		return d_gamePhase;
+	}
+
```
Comment (amyhcy) [inline src/warzone/src/warzone/service/GameEngine.java:73]: missing javadoc

File: src/warzone/src/warzone/service/GameEngine.java [hunk 4]
```diff
@@ -36,25 +56,47 @@ public static GameEngine getGameEngine(GameContext p_gameContext) {
...
+	 * State object of the GameEngine 
 	 */
-	public static void main(String[] args) throws IOException {
-
-		GameContext l_gameContext = GameContext.getGameContext();
-		RouterService l_routerService =  RouterService.getRouterService(l_gameContext);
-		CommandService l_commandService =  CommandService.getCommandService(l_gameContext);
-		
+	private Phase d_gamePhase ;
+
```
Comment (amyhcy) [inline src/warzone/src/warzone/service/GameEngine.java:65]: missing javadoc

File: src/warzone/src/warzone/service/RouterService.java [hunk 5]
```diff
@@ -19,13 +20,16 @@
 	private static RouterService ROUTER_SERVICE;
 	
 	private GameContext d_gameContext;
-
+	private GameEngine d_gameEngine;
+	private Phase d_gamePhase;
 	/**
 	 * the constructor of it, only can be used inside this class.
 	 * @param p_gameContext the current game context
```
Comment (amyhcy) [inline src/warzone/src/warzone/service/RouterService.java]: wrong parm

File: src/warzone/src/warzone/service/RouterService.java [hunk 6]
```diff
@@ -34,150 +38,88 @@ private RouterService(GameContext p_gameContext) {
 	 * @param p_gameContext the game context instance
```
Comment (amyhcy) [inline src/warzone/src/warzone/service/RouterService.java]: wrong parm

File: src/warzone/src/warzone/view/HelpView.java [hunk 7]
```diff
@@ -29,23 +31,24 @@ public static void printWelcome() {
      * This method will print current status of the game from the game context.
      * @param p_gameContext the current game context
```
Comment (amyhcy) [inline src/warzone/src/warzone/view/HelpView.java]: wrong pram

**Comment by fzuray** [comment]
fixed all the java doc

**Pull request merged**